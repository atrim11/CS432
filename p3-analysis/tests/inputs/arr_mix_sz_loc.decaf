int arr[3];

def int main() {
int b;
int c;
int d;
int z;
z = 0;
d = 5;
b=arr[z];
b = arr[-z];
b=arr[3];
b=arr[c];
b= arr[d];
return b;
}
