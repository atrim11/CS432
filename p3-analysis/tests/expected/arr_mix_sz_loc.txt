Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 main : () -> int
 arr : int [3]

  FuncDecl name="main" return_type=int parameters={} [line 3]
  SYM TABLE:

    Block [line 3]
    SYM TABLE:
     b : int
     c : int
     d : int
     z : int

